### PR TITLE
chore: Merges Azure KMS Encryption at Rest Private Endpoint feature to master

### PR DESCRIPTION
**Pull request by maastha**
## Description

Merges Azure KMS Encryption at Rest Private Endpoint feature to master

Link to any related issue(s): [CLOUDP-267671](https://jira.mongodb.org/browse/CLOUDP-267671)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .github/workflows/code-health.yml [hunk 1]
```diff
@@ -70,11 +70,15 @@ jobs:
...
+        run: make generate-doc resource_name=push_based_log_export
       - name: Doc for search_deployment
-        run: export resource_name=search_deployment && make generate-doc
+        run: make generate-doc resource_name=search_deployment
+      - name: Doc for encryption_at_rest
+        run: make generate-doc resource_name=encryption_at_rest
+      - name: Doc for encryption_at_rest_private_endpoint
+        run: make generate-doc resource_name=encryption_at_rest_private_endpoint
       - name: Doc for project_ip_addresses
         run: export resource_name=project_ip_addresses && make generate-doc
```
Comment (lantoli) [inline .github/workflows/code-health.yml]: use the same way for all resources, I understand we have to do it if Makefile target was changed to introduce the param, so I think  `project_ip_addresses` needs to be changed

File: internal/common/conversion/type_conversion.go [hunk 2]
```diff
@@ -62,3 +62,8 @@ func IsStringPresent(strPtr *string) bool {
 func MongoDBRegionToAWSRegion(region string) string {
 	return strings.ReplaceAll(strings.ToLower(region), "_", "-")
 }
+
+// AWSRegionToMongoDBRegion converts region in us-east-1-like format to US_EAST_1-like
+func AWSRegionToMongoDBRegion(region string) string {
```
Comment (lantoli) [inline internal/common/conversion/type_conversion.go:67]: `conversion.ValRegion` does something similar, is it possible to unifiy both functions so we only have one logic to transform between region formats?

File: internal/common/retrystrategy/retry_state.go [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
 const (
-	RetryStrategyPendingState   = "PENDING"
-	RetryStrategyCompletedState = "COMPLETED"
-	RetryStrategyErrorState     = "ERROR"
-	RetryStrategyPausedState    = "PAUSED"
-	RetryStrategyUpdatingState  = "UPDATING"
-	RetryStrategyIdleState      = "IDLE"
-	RetryStrategyDeletedState   = "DELETED"
+	RetryStrategyPendingState    = "PENDING"
+	RetryStrategyCompletedState  = "COMPLETED"
```
Comment (lantoli) [inline internal/common/retrystrategy/retry_state.go:5]: these consts are in `retrystrategy`package so starting with `RetryStrategy... ` seems redundant

**Review by lantoli — APPROVED** [review]

File: .github/workflows/code-health.yml (hunk 1 above)
Comment (maastha) [reply .github/workflows/code-health.yml]: updated

File: internal/common/retrystrategy/retry_state.go (hunk 3 above)
Comment (maastha) [reply internal/common/retrystrategy/retry_state.go:5]: will address this in follow-up PRs as would not like to make this PR bigger for now

Comment (lantoli) [reply internal/common/retrystrategy/retry_state.go:5]: sure, thanks

**Review by AgustinBettati — APPROVED** [review]
LGTM, we can consider merging with a merge commit if we want to preserve more specific git history.

File: docs/data-sources/encryption_at_rest.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
+# Data Source: mongodbatlas_encryption_at_rest
+
+`mongodbatlas_encryption_at_rest` describes encryption at rest configuration for an Atlas project with one of the following providers:
+
+[Amazon Web Services Key Management Service](https://docs.atlas.mongodb.com/security-aws-kms/#security-aws-kms)
+[Azure Key Vault](https://docs.atlas.mongodb.com/security-azure-kms/#security-azure-kms)
+[Google Cloud KMS](https://docs.atlas.mongodb.com/security-gcp-kms/#security-gcp-kms)
+
+
+~> **IMPORTANT** Atlas encrypts all cluster storage and snapshot volumes, securing all cluster data on disk: a concept known as encryption at rest, by default.
```
Comment (pierwill) [inline docs/data-sources/encryption_at_rest.md]:
```suggestion
~> **IMPORTANT** By default, Atlas enables encryption at rest for all cluster storage and snapshot volumes.
```

**Review by pierwill — APPROVED** [review]
Approved with one suggestion.

File: docs/data-sources/encryption_at_rest.md (hunk 4 above)
Comment (maastha) [reply docs/data-sources/encryption_at_rest.md]: updated this in the PR

**Pull request merged**